### PR TITLE
Bugfix: RPC/Mining: Use pre-segwit sigops and limits, when working with non-segwit GBT clients

### PR DESCRIPTION
**Pull request by luke-jr**

### CODE REVIEW

**Comment by sipa** [comment]
utACK 4292a752c7eb59ef2cbee70b5cd186eedf16d0dc

**Comment by luke-jr** [comment]
Rebased. At this point, it might make sense to just drop support for pre-segwit miners, but this fix is simple, so it makes sense to get in (and backportable) first.

File: test/functional/feature_segwit.py [hunk 1]
```diff
@@ -225,6 +225,13 @@ def run_test(self):
         # Now try calling getblocktemplate() without segwit support.
         template = self.nodes[0].getblocktemplate()
 
+        # Check that we got a pre-Segwit template
+        assert(template['sizelimit'] == 1000000)
```
Comment (jnewbery) [inline test/functional/feature_segwit.py]: nit: prefer `assert_equal()` for asserting equality

File: test/functional/feature_segwit.py [hunk 2]
```diff
@@ -225,6 +225,13 @@ def run_test(self):
...
+        assert('weightlimit' not in template)
```
Comment (jnewbery) [inline test/functional/feature_segwit.py]: nit: assert is a statement, not a function. Please remove parens.

**Review by jnewbery — COMMENTED** [review]
I agree that we should just drop support for non-segwit miners at this point. Doesn't seem worth the trouble of making this change.

**Comment by MarcoFalke** [comment]
This has been "fixed" by 
*    #14811 (Mining: Enforce that segwit option must be set in GBT by jnewbery)

**Pull request closed without merging**